pkg/install/v1alpha1/jiva_volume.go

```diff
@@ -277,30 +289,25 @@ metadata:
   name: jiva-volume-isvalidversion-default
 spec:
   meta: |
-    id: is070jivavolume
-    runNamespace: {{.Volume.runNamespace}}
+    id: is082jivavolume
+    runNamespace: {{ .Volume.runNamespace }}
     apiVersion: v1
     kind: Service
     action: list
     options: |-
-      labelSelector: vsm={{ .Volume.owner }}
+      labelSelector: openebs.io/controller-service=jiva-controller-svc,openebs.io/version=0.8.2
   post: |
-    {{- jsonpath .JsonResult "{.items[*].metadata.name}" | trim | saveAs "is070jivavolume.name" .TaskResult | noop -}}
-    {{- .TaskResult.is070jivavolume.name | empty | not | versionMismatchErr "is not a jiva volume of 0.7.0 version" | saveIf "is070jivavolume.versionMismatchErr" .TaskResult | noop -}}
+    {{- jsonpath .JsonResult "{.items[*].metadata.name}" | trim | saveAs "is082jivavolume.name" .TaskResult | noop -}}
+    {{- .TaskResult.is082jivavolume.name | empty | not | versionMismatchErr "is not a jiva volume of 0.8.2 version" | saveIf "is082jivavolume.versionMismatchErr" .TaskResult | noop -}}
```

Comment:
this fallback approach is not correct.
Ideally, runtask should make its CAST to fallback if it is not jiva volume of latest. Fallback would be 0.8.2
If its not 0.8.2, again, CAST should fallback to 0.7
But, here, we are checking whether given pv is 0.8.2, and doing fallback if so. So, if the volume is either latest or older than 0.8.2, fallback won't get executed which is problem for volumes older than 0.8.2.

Reply:
right @vishnuitta the fallback logic is not simple here. We need to consider a range of versions i.e. 0.7.1 till 0.8.2 that should use 0.8.2 RunTasks as fallback logic.
We should also see if we can really determine the actual version from any old volume?
Currently, we use some override script that forcibly sets/overrides the actual version to the one set in the script. Again this overriding was a wrong decision IMO.

On the whole, we need to re-look at upgrade & version.